### PR TITLE
feat(new-rule): ibm-prefer-token-pagination

### PR DESCRIPTION
**Pull request by dpopp07**
Adds a rule to check paginated list operations for offset/limit style pagination. If found, a warning is given encouraging the user to use token-based pagination instead.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Dependencies have been updated as needed
- [x] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

### CODE REVIEW

File: docs/ibm-cloud-rules.md [hunk 1]
```diff
@@ -3770,9 +3777,9 @@ paths:
 </tr>
 <tr>
 <td valign=top><b>Description:</b></td>
-<td>Operations that return a 412 status code must support at least one of the following header parameters: <code>If-Match</code>, <code>If-None-Match</code>, <code>If-Modified-Since</code>, <code>If-Unmodified-Since</code></td>.
+<td>Operations that return a 412 status code must support at least one of the following header parameters: <code>If-Match</code>, <code>If-None-Match</code>, <code>If-Modified-Since</code>, <code>If-Unmodified-Since</code>.
 For more details, please see the API Handbook section on
-<a href="https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-headers#conditional-headers">Conditional headers</a>.
+<a href="https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-headers#conditional-headers">Conditional headers</a>.</td>
```
Comment (dpopp07) [inline docs/ibm-cloud-rules.md:3782]: I believe this was a typo - the formatting is odd in the doc. So I went ahead and fixed it

File: packages/ruleset/src/functions/pagination-style.js [hunk 2]
```diff
@@ -37,96 +45,35 @@ module.exports = function (pathObj, _opts, context) {
...
-
-  // Check to see if the operation defines an "offset" query param.
-  const offsetParamIndex = params.findIndex(
-    param => param.name === 'offset' && param.in === 'query'
-  );
-
-  // If the operation doesn't define a page token-type query param or an "offset" query param,
-  // then bail out now as pagination isn't supported by this operation.
-  if (pageTokenParamIndex < 0 && offsetParamIndex < 0) {
-    logger.debug(`${ruleId}: No start or offset query param!`);
```
Comment (dpopp07) [inline packages/ruleset/src/functions/pagination-style.js:120]: All of this code was moved so we could have one function to determine if a path has a paginated list operation or not

File: packages/validator/test/cli-validator/mock-files/oas3/clean-with-tabs.yml [hunk 3]
```diff
@@ -31,13 +31,15 @@ paths:
 						format: int32
 						default: 10
 						maximum: 100
-				- name: offset
+				- name: start
```
Comment (padamstx) [inline packages/validator/test/cli-validator/mock-files/oas3/clean-with-tabs.yml:34]: I guess we chose the wrong style of pagination for our "clean" API definition :joy:

**Review by padamstx — APPROVED** [review]
LGTM!

**Review by padamstx — APPROVED** [review]

**Pull request merged**